config-svc/pkg/metacfg/manager.go

```diff
@@ -0,0 +1,79 @@
+package metacfg
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+type ConfigManager interface {
+	Get() Config
+	Set(config *Config) error
+}
+
+type EphemeralConfigManager struct {
+	value *Config
+	mux   sync.RWMutex
+}
+
+func (m *EphemeralConfigManager) Get() Config {
+	m.mux.RLock()
```

Comment:
I think generally the pattern is to immediately defer the unlock afterwards. I know we only have one statement here but it is good to follow.